### PR TITLE
Ready: OTP 18 support

### PR DESCRIPTION
**Pull request by lukebakken**
See original PR #234 

Depends on basho/riak_pb#160

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+    warnings_as_errors,
+    {platform_define, "^[0-9]+", namespaced_types},
+    {platform_define, "^18", deprecated_now}
+]}.
 {deps, [
-        {riak_pb, "2.1.0.2", {git, "git://github.com/basho/riak_pb", {tag, "2.1.0.2"}}}
-       ]}.
-{edoc_opts, [{stylesheet_file, "priv/edoc.css"},
-             {preprocess, true}]}.
+    {riak_pb, "2.1.0.6", {git, "git://github.com/basho/riak_pb", {branch, "features/lrb/otp-18"}}}
```
Comment (lukebakken) [inline rebar.config]: Note: will change this to updated `riak_pb` tag once basho/riak_pb#160 is merged.

File: .travis.yml [hunk 2]
```diff
@@ -6,8 +6,6 @@ notifications:
...
   email: eng@basho.com
 otp_release:
-  - R16B02
-  - R16B01
-  - R15B03
-  - R15B02
-  - R15B01
+  - 18.1
+  - 17.5
+  - R16B03
```
Comment (javajolt) [inline .travis.yml]: Is this true? AFAIK we are still on R16B02 as our supported version. Of course it's a Basho special...

**Comment by javajolt** [comment]
@lukebakken I had meant to YOLO merge the riak_pb PR because Bors was stuck on that project.  AFAIK bors is still working fine on the Erlang client.

File: .travis.yml (hunk 2 above)
Comment (lukebakken) [inline .travis.yml]: I'll put that version back.

**Pull request merged**